src/Swarm/Game/Location.hs

```diff
@@ -139,6 +141,28 @@ cardinalDirs =
 toDirection :: Heading -> Maybe Direction
 toDirection v = M.lookup v cardinalDirs
 
+-- | Example:
+--      DWest `relativeTo` DSouth == DRight
+relativeTo :: AbsoluteDir -> AbsoluteDir -> RelativeDir
+relativeTo targetDir referenceDir =
+  [DForward, DLeft, DBack, DRight] !! indexDiff
+ where
+  enumCount = length (Util.listEnums :: [AbsoluteDir])
+  indexDiff = (((-) `on` fromEnum) targetDir referenceDir + enumCount) `mod` enumCount
```

Comment:
```suggestion
  indexDiff = ((-) `on` fromEnum) targetDir referenceDir `mod` enumCount
```
Adding `enumCount` before doing the `mod` is a workaround only needed for languages that do not have a sensible semantics for `mod` (*e.g.* Java, C).

Reply:
Ideal for my use case (if not sensible) would be to return a positive value even if the left operand is negative.  But that's not what the Haskell `mod` operator does: https://replit.com/@KarlOstmo/LostWretchedDatamining

Is that what you were referring to?

Reply:
No, that *is* what the Haskell `mod` function does.  ``-5 `mod` 3`` parses as ``-(5 `mod` 3)``.  Try ``(-5) `mod` 3``. :smile:

Reply:
When `b` is positive, `mod` always satisfies ``0 <= a `mod` b < b`, and ``a = (a `div` b) * b + (a `mod` b)``.

Reply:
> No, that _is_ what the Haskell `mod` function does. `` -5 `mod` 3 `` parses as `` -(5 `mod` 3) ``. Try `` (-5) `mod` 3 ``. 😄

HA! Oh wow.

Reply:
I feel like there should be a lint rule that warns on misleading use of whitespace/lack of parenthesization with the `-` operator

Reply:
There 100% should be. Negation in Haskell is a bit strange... also cf how you have to write `subtract 2` instead of `(-2)` if you want an operator section.

Reply:
https://github.com/ndmitchell/hlint/issues/1484